app/models/king.rb

```diff
@@ -1,5 +1,48 @@
 class King < Piece
 
+  def valid_move?(new_x, new_y)
+    #checks if the move that the king is making is valid
+    if position_exist?(new_x, new_y) && pass_king_rules?(new_x, new_y)
+      return true
+    else
+      return false
+    end
+  end
+
+  def position_exist?(new_x, new_y)
+    #checks if position that king is moving to exists
+    if new_x < 0 || new_x > 7 || new_y < 0 || new_y > 7
+      return false
+    else
+      return true
+    end
+  end
+
+  def allowed_positions
```

Comment:
@ruishine that's a great idea but I have nooo idea how to do that lol I'm open to trying it out though, maybe we can pair program it?